File: SDKs/googleapis.com/firebasedynamiclinks/v1/python/src/sdk/models/shared/createmanagedshortlinkresponse.py
import dataclasses
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import managedshortlink as shared_managedshortlink
from ..shared import dynamiclinkwarning as shared_dynamiclinkwarning


@dataclass_json
@dataclasses.dataclass
class CreateManagedShortLinkResponse:
    r"""CreateManagedShortLinkResponse
    Response to create a short Dynamic Link.
    """
    
    managed_short_link: Optional[shared_managedshortlink.ManagedShortLink] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('managedShortLink') }})
    preview_link: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('previewLink') }})
    warning: Optional[list[shared_dynamiclinkwarning.DynamicLinkWarning]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('warning') }})
    

File: SDKs/googleapis.com/firebasedynamiclinks/v1/java/lib/src/main/java/openapisdk/models/shared/CreateManagedShortLinkResponse.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * CreateManagedShortLinkResponse
 * Response to create a short Dynamic Link.
**/
public class CreateManagedShortLinkResponse {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("managedShortLink")
    public ManagedShortLink managedShortLink;
    public CreateManagedShortLinkResponse withManagedShortLink(ManagedShortLink managedShortLink) {
        this.managedShortLink = managedShortLink;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("previewLink")
    public String previewLink;
    public CreateManagedShortLinkResponse withPreviewLink(String previewLink) {
        this.previewLink = previewLink;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("warning")
    public DynamicLinkWarning[] warning;
    public CreateManagedShortLinkResponse withWarning(DynamicLinkWarning[] warning) {
        this.warning = warning;
        return this;
    }
}